numpy/core/numeric.py

```diff
@@ -58,6 +58,22 @@
     '_get_promotion_state', '_set_promotion_state']
 
 
+# TODO: Remove this wrapper when fastCopyAndTranspose deprecation expires
+def fastCopyAndTranspose(a):
+    """
+    .. deprecated:: 1.24
+
+       fastCopyAndTranspose is deprecated and will be removed. Use the copy
+       and transpose methods instead, e.g. ``arr.T.copy()``
+    """
+    warnings.warn(
+        "fastCopyAndTranspose is deprecated. Use ``arr.T.copy()`` instead",
+        DeprecationWarning,
+        stacklevel=2,
+    )
+    return _fastCopyAndTranspose(a)
```

Comment:
Does this now emit two warnings: one from the python function and one from the C function?

Reply:
Ahh, no, this no longer calls the C function. Sorry.

Reply:
What does it call then?  The linalg version?  That seems to do something slightly different since it accepts many arrays?

Reply:
I'm a bit confused about the general procedure for deprecating something from the C-API. Does it even make sense for users to get C-API warnings in Python code? At runtime? In principle that's what would happen here (assuming the user called `_fastCopyAndTranspose` and I set the stacklevel for the warning correctly) but now I'm doubting whether this is the desired outcome.

I'm going to mark this for triage-review in the hope that a quick discussion will help clear things up for me!